Library/Homebrew/cmd/cleanup.rb

```diff
@@ -83,7 +83,9 @@ def cleanup_cache
         next
       end
 
-      if f.version > version || ARGV.switch?('s') && !f.installed? || bottle_file_outdated?(f, file)
+      file_is_stale = PkgVersion === version ? f.pkg_version > version : f.version > version
+
+      if file_is_stale || ARGV.switch?('s') && !f.installed? || bottle_file_outdated?(f, file)
```

Comment:
I didn't change anything so far as I realize this whole method need to do a serious change. (e.g. better matching for name, handling resources etc.) So I intend to address this in future PR.